### ISSUE TITLE
Feature Request: Allow customization to open images in external viewer.

### ISSUE THREAD

**Issue by michelk** [jrblevin/markdown-mode#383]
Images associated with image links may be displayed inline in the buffer. This is not possible when not using a GUI (`emacsclient`). `C-c C-0` then opens a useless text version. 

It should be possible to define an external program like `eog` to open the image. In org-mode this is done by defining ` (add-to-list 'org-file-apps '("\\.png\\'" . "eog %s"))`

It would be great, if this would also be possible in `markdown-mode`.

Issue first raised in https://github.com/syl20bnr/spacemacs/issues/11973

**Comment by saf-dmitry** [comment]
You can delegate opening external links to Org mode by following command:
```elisp
(defun markdown-org-follow-link-at-point ()
  "Open external Markdown link as if it was in Org mode.
See variable `org-file-apps' for setting external applications
for opening links in a document."
  (interactive)
  (if (and (markdown-link-p)
           (not (string-prefix-p "#" (markdown-link-url))))
      (org-open-link-from-string (format "[[%s]]" (markdown-link-url)))
    (user-error "Point is not at an external Markdown link or URL")))
```

You can bind this command to `C-c C-o` or another suitable key combination.

**Comment by saf-dmitry** [comment]
@michelk You can delegate opening external Markdown links to Org mode by using the following command, wich you can bind to `C-c C-o` or another suitable key combination:
```elisp
(defun markdown-org-follow-link-at-point ()
  "Open external Markdown link as if it was in Org mode.
See variable `org-file-apps' for setting external applications
for opening links in a document."
  (interactive)
  (if (and (markdown-link-p)
           (not (string-prefix-p "#" (markdown-link-url))))
      (org-open-link-from-string (format "[[%s]]" (markdown-link-url)))
    (user-error "Point is not at an external Markdown link or URL")))
```

**Comment by saf-dmitry** [comment]
@michelk For the time being you can delegate opening external Markdown links to Org mode by using the following command, which you can bind to `C-c C-o` or another suitable key combination:
```elisp
(defun markdown-org-follow-link-at-point ()
  "Open external Markdown link as if it was in Org mode.
See variable `org-file-apps' for setting external applications
for opening links in a document."
  (interactive)
  (if (and (markdown-link-p)
           (not (string-prefix-p "#" (markdown-link-url))))
      (org-open-link-from-string (format "[[%s]]" (markdown-link-url)))
    (user-error "Point is not at an external Markdown link or URL")))
```

**Comment by syohex** [comment]
I have merged #467. Please check latest version and set `markdown-open-image-command` customize variable
